service/src/main/java/com/epam/esm/util/validator/impl/UserValidator.java

```diff
@@ -0,0 +1,27 @@
+package com.epam.esm.util.validator.impl;
+
+import com.epam.esm.entity.User;
+
+import com.epam.esm.util.validator.Validator;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserValidator implements Validator<User> {
+
+    @Override
+    public boolean isValid(User user) {
+        return isPresent(user);
+    }
+
+    public boolean isPresent(User user){
+        if(user.getNickName().isEmpty() && user.getEmail().isEmpty()
+        && user.getPassword().isEmpty()){
+            return false;
+        } else {
+            return true;
+        }
+    }
```

Comment:
🤨🤨🤨
Based on your written logic user is valid when he has nickName or email or password.
So,
`
 public boolean isValid(User user) {
        return !user.getNickName().isEmpty() || !user.getEmail().isEmpty() || user.getPassword().isEmpty());
}
` will be much easier!

Reply:
Fixed
Thanks